lib/Raven/Client.php

```diff
@@ -231,76 +110,67 @@ class Client
     /**
      * @var bool
      */
-    protected $_shutdown_function_has_been_set;
+    protected $_shutdown_function_has_been_set = false;
+
+    /**
+     * @var Configuration The client configuration
+     */
+    protected $config;
 
-    public function __construct($options_or_dsn = null, $options = [])
+    /**
+     * Constructor.
+     *
+     * @param Configuration $config The client configuration
+     */
+    public function __construct(Configuration $config)
```

Comment:
That would make sense if we wanted to preserve BC; but since we are breaking it in a million other ways, it will just clutter the codebase with two different ways of passing options through. And the array-way will not have all the safeties that an object will have, with typehints, getters, absent setters when dangerous etc....